### DISCUSSION TITLE
Skin float cap

### DISCUSSION THREAD

**Discussion by quebeh** [ByMykel/CSGO-API#102] [Q&A] [answered]
Greetings, I'm trying to get the float cap of every skin that is currently in CS2.
As you know, some items have different float caps that limits what the possible float for an item is.
For example, the AWP Lightning Strike can only be Factory New or Minimal Wear, with float cap of 0.00-0.08, while gloves has a float cap of 0.06-0.8.

I am trying to retrieve that data and store it to a JSON file, preferably only when I need it. Is there any way of doing that currently?
All I've found online about this is from posts from 3 years ago, 9 years ago... and they all suggest looking at the game files, `csgo/scripts/items/items_game.txt`. Though this file does not exist anymore. 

Any ideas on how to do this?

**Comment by ByMykel** [comment, marked as answer]
Hi. Actually I am already doing that here [skins_not_grouped.json](https://bymykel.github.io/CSGO-API/api/en/skins_not_grouped.json) check `min_float` and `max_float`.

If you want to do it by yourself you have to take the items_game.txt and parse it to json like I am doing on my [repo](https://github.com/ByMykel/counter-strike-file-tracker/tree/main/static).

Then go throught all that data and map your own JSON like you want (check [these files](https://github.com/ByMykel/CSGO-API/tree/main/services) to know how I do for my API).

The easiest path to archive your goal is to take `skins_not_grouped.json` and map it removing everything but the attributes you need to get something like:

```js
[
 {
        "name": "Desert Eagle | Urban DDPAT (Factory New)",
        "min_float": 0.06,
        "max_float": 0.8,
    },
    {
        "name": "Desert Eagle | Urban DDPAT (Minimal Wear)",
        "min_float": 0.06,
        "max_float": 0.8,
    },
    {
        "name": "Desert Eagle | Urban DDPAT (Field-Tested)",
        "min_float": 0.06,
        "max_float": 0.8,
    },
]
```

Also I have [skins.json](https://bymykel.github.io/CSGO-API/api/en/skins.json) that is the same as `skins_not_grouped.json` but the data is grouped by skins.

Let me know if you have more questions.

EDIT: You also already have in `skins.json` the possible wears for a skin:
```js
{
  wears": [
    {
        "id": "SFUI_InvTooltip_Wear_Amount_0",
        "name": "Factory New"
    },
    {
        "id": "SFUI_InvTooltip_Wear_Amount_1",
        "name": "Minimal Wear"
    },
    {
        "id": "SFUI_InvTooltip_Wear_Amount_2",
        "name": "Field-Tested"
    },
    {
        "id": "SFUI_InvTooltip_Wear_Amount_3",
        "name": "Well-Worn"
    },
    {
        "id": "SFUI_InvTooltip_Wear_Amount_4",
        "name": "Battle-Scarred"
    }
],
}
```

**Reply by quebeh** [reply]
Ah, that's helpful. I went through the open-source tracking repo of CS2 and found `items_game.txt` myself, though the skin names are very messy and does not use the names shown in-game. I figured I had to use the `paint_index` and dump it all into JSON. But `skins_not_grouped.json` from this repo is definitely very convenient and saves time. Thank you very much!
